### PR TITLE
[EAGLE-782] Upgrade org.codehaus.jackson to com.fasterxml.jackson

### PR DESCRIPTION
**Pull request by baibaichen**
http://issues.apache.org/jira/browse/EAGLE-782
Upgrade org.codehaus.jackson to com.fasterxml.jackson

### CODE REVIEW

**Comment by baibaichen** [comment]
@haoch  please review this PR,  if it is ok, I will create JIRA.

File: eagle-core/eagle-alert-parent/eagle-alert/alert-engine/src/test/java/org/apache/eagle/alert/engine/statecheck/TestStateCheckPolicy.java [hunk 1]
```diff
@@ -130,7 +130,7 @@ private void sendData(AlertBolt alertBolt, Map<String, StreamDefinition> definit
     private static ObjectMapper mapper = new ObjectMapper();
 
     static {
-        mapper.configure(DeserializationConfig.Feature.FAIL_ON_UNKNOWN_PROPERTIES, false);
+        mapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
```
Comment (baibaichen) [inline eagle-core/eagle-alert-parent/eagle-alert/alert-engine/src/test/java/org/apache/eagle/alert/engine/statecheck/TestStateCheckPolicy.java]: the only place I need modify code

**Comment by haoch** [comment]
@baibaichen LGTM, please go ahead to create JIRA.

**Comment by baibaichen** [comment]
@haoch , I already added [EAGLE-782](https://issues.apache.org/jira/browse/EAGLE-782).  let me fix conflicts first.

**Comment by jhsenjaliya** [comment]
FYI, I still find some reference to org.codehaus.jackson.* in following files.

1. eagle-core/eagle-alert-parent/eagle-alert/alert-service/src/main/webapp/WEB-INF/web.xml
2. eagle-webservice/src/main/webapp/WEB-INF/web.xml
3. eagle-server/src/main/webapp/WEB-INF/web.xml
4. eagle-core/eagle-query/eagle-service-base/src/main/webapp/WEB-INF/web.xml

otherwise, LGTM.

**Comment by baibaichen** [comment]
@jhsenjaliya  I updated web.xml and resolve conflicts, thanks!

By the way I find some class will refer org.codehaus.jettison.json.JSONException and org.codehaus.jettison.json.JSONObject

Does it matter?

**Comment by jhsenjaliya** [comment]
@baibaichen, Yes, those should be fine, at least not related to this change.
Thanks, it LGTM now.

**Comment by baibaichen** [comment]
@haoch  please re-review this patch.  

Since we will release the new version at Feb., it would be low risk to merge this patch now.

**Pull request closed without merging**